modules/auxiliary/scanner/http/options.rb

```diff
@@ -14,60 +14,76 @@ class MetasploitModule < Msf::Auxiliary
 
   def initialize
     super(
-      'Name'        => 'HTTP Options Detection',
+      'Name' => 'HTTP Options Detection',
       'Description' => 'Display available HTTP options for each system',
-      'Author'       => ['CG'],
-      'License'     => MSF_LICENSE,
-      'References' =>
+      'Author' => ['CG'],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['CVE', '2005-3398'], # HTTP Trace related
+        ['CVE', '2005-3498'], # HTTP Trace related
+        ['OSVDB', '877'],
+        ['BID', '11604'],
+        ['BID', '9506'],
+        ['BID', '9561'],
+        ['URL', 'https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS']
+      ]
+    )
+    register_options(
       [
-        [ 'CVE', '2005-3398'],
-        [ 'CVE', '2005-3498'],
-        [ 'OSVDB', '877'],
-        [ 'BID', '11604'],
-        [ 'BID', '9506'],
-        [ 'BID', '9561']
+        OptString.new('TARGETURI', [true, 'URI to test', '/']),
       ]
     )
   end
 
   def run_host(target_host)
+    res = send_request_cgi({
+      'uri' => datastore['TARGETURI'],
+      'method' => 'OPTIONS'
+    })
+    return unless res
+    return if res.nil?
```

Comment:
These two lines are pretty much equivalent? I imagine we can skip the `.nil? case and keep the first line.